src/hooks/useTokeList.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react'
+import { tokenListApi } from 'api'
+import { TokenDetails } from 'types'
+import useSafeState from './useSafeState'
+
+// for stable reference
+// to avoid updates on setState([])
+const emptyArray: TokenDetails[] = []
+
+export const useTokenList = (networkId?: number): TokenDetails[] => {
+  const [tokens, setTokens] = useSafeState(networkId === undefined ? emptyArray : tokenListApi.getTokens(networkId))
+
+  useEffect(() => {
+    if (networkId === undefined) return setTokens(emptyArray)
+    return tokenListApi.subscribe(setTokens)
```

Comment:
:thinking: 
When the hook umounts, shouldn't it return `tokenListApi.unsubscribe`?

Reply:
```ts
public subscribe(callback: SubscriptionCallback<State>): Command {
    this._subscriptions.push(callback)

    return (): void => this.unsubscribe(callback)
  }
```

Reply:
:brainfreeze: